flow/cmd/mirror_status.go

```diff
@@ -0,0 +1,256 @@
+		// The clone table jobs that are children of the CDC snapshot flow
+		// do not have a config entry, so allow this to be nil.
+		Config:     h.getQRepConfigFromCatalog(req.FlowJobName),
+		Partitions: partitionStatuses,
+	}, nil
+}
+
+// getPartitionStatus returns the status of a partition uuid.
+func (h *FlowRequestHandler) getPartitionStatus(
+	ctx context.Context,
+	partitionUUID string,
+) (*protos.PartitionStatus, error) {
+	partitionStatus := &protos.PartitionStatus{
+		PartitionId: partitionUUID,
+	}
+
+	var startTime pgtype.Timestamp
+	var endTime pgtype.Timestamp
+	var numRows pgtype.Int4
+
+	q := "SELECT start_time, end_time, rows_in_partition FROM peerdb_stats.qrep_partitions WHERE partition_uuid = $1"
+	err := h.pool.QueryRow(ctx, q, partitionUUID).Scan(&startTime, &endTime, &numRows)
+	if err != nil {
+		return nil, fmt.Errorf("unable to query qrep partition - %s: %w", partitionUUID, err)
+	}
+
+	if startTime.Valid {
+		partitionStatus.StartTime = timestamppb.New(startTime.Time)
+	}
+
+	if endTime.Valid {
+		partitionStatus.EndTime = timestamppb.New(endTime.Time)
+	}
+
+	if numRows.Valid {
+		partitionStatus.NumRows = numRows.Int32
+	}
+
+	return partitionStatus, nil
+}
+
+func (h *FlowRequestHandler) getPartitionUUIDs(
+	ctx context.Context,
+	flowJobName string,
+) ([]string, error) {
+	rows, err := h.pool.Query(ctx,
+		"SELECT partition_uuid FROM peerdb_stats.qrep_partitions WHERE flow_name = $1", flowJobName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to query qrep partitions: %w", err)
+	}
+	defer rows.Close()
+
+	partitionUUIDs := []string{}
+	for rows.Next() {
+		var partitionUUID pgtype.Text
+		if err := rows.Scan(&partitionUUID); err != nil {
+			return nil, fmt.Errorf("unable to scan partition row: %w", err)
+		}
+		partitionUUIDs = append(partitionUUIDs, partitionUUID.String)
+	}
+
+	return partitionUUIDs, nil
+}
+
+func (h *FlowRequestHandler) getFlowConfigFromCatalog(
+	flowJobName string,
+) (*protos.FlowConnectionConfigs, error) {
+	var configBytes sql.RawBytes
+	var err error
+	var config protos.FlowConnectionConfigs
+
+	err = h.pool.QueryRow(context.Background(),
+		"SELECT config_proto FROM flows WHERE name = $1", flowJobName).Scan(&configBytes)
+	if err != nil {
+		return nil, fmt.Errorf("unable to query flow config from catalog: %w", err)
+	}
+
+	err = proto.Unmarshal(configBytes, &config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal flow config: %w", err)
+	}
+
+	return &config, nil
+}
+
+func (h *FlowRequestHandler) getQRepConfigFromCatalog(
+	flowJobName string,
+) *protos.QRepConfig {
+	var configBytes []byte
+	var err error
+	var config protos.QRepConfig
+
+	err = h.pool.QueryRow(context.Background(),
+		"SELECT config_proto FROM flows WHERE name = $1", flowJobName).Scan(&configBytes)
+	if err != nil {
+		logrus.Warnf("unable to query qrep config from catalog: %s", err.Error())
+		return nil
+	}
+
+	err = proto.Unmarshal(configBytes, &config)
+	if err != nil {
+		return nil
```

Comment:
Any reason why we don't want to return an error here like in the above function?
```go
if err != nil {
    return nil, fmt.Errorf("unable to unmarshal qrep config: %w", err)
}
```

Reply:
in case where initial load is done for cdc we do not add the entry to flows table